src/shims/time.rs

```diff
@@ -1,8 +1,71 @@
+use std::sync::atomic::AtomicU64;
+use std::sync::Arc;
 use std::time::{Duration, Instant, SystemTime};
 
+use rustc_data_structures::sync::Ordering;
+
 use crate::concurrency::thread::Time;
 use crate::*;
 
+#[derive(Debug)]
+pub enum TimeAnchor {
+    Host(Instant),
+    Virtual(Arc<AtomicU64>),
```

Comment:
Why is there an `Arc`? There should be a comment somewhere explaining the design. But I don't think an `Arc` should be needed.